### PR TITLE
read postgres and redis ports from config file

### LINKED ISSUE #154: Read postgres and redis ports from config file
I think this would be better to move all this kind of data to config files.

### LINKED ISSUE #162: Read database and redis connection parameters from config

### PR DESCRIPTION
**Pull request by Prytu**
fixes #154 
fixes #162

### CODE REVIEW

File: backend/config/interfaces.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// Provides DB access configuration
+type DBAccessConfigProvider interface {
+	GetDBUsername() string
+	GetDBPassword() string
+	GetDBHost() string
+	GetDBPort() string
+}
+
+// Provides Redis access configuration
+type RedisAccessConfigProvider interface {
```
Comment (VirrageS) [inline backend/config/interfaces.go]: I think that this should be in `CacheConfigProvider` and functions should be named `GetCachePassword` etc. We want `Cache` to be generic for some strategies like: `Redis`, `Memcache`, `DummyCache`.

File: backend/config/config.go [hunk 2]
```diff
@@ -10,37 +10,22 @@ import (
...
-
-// Stores global configuration of the system
+// Stores global configuration of the system. Implements all config interfaces
 type Configuration struct {
 	secretKey                  []byte
 	authTokenValidityPeriod    int
 	refreshTokenValidityPeriod int
 	cacheExpirationTime        time.Duration
+
+	DBUsername string
```
Comment (VirrageS) [inline backend/config/config.go]: I think that there should be new struct `DatabaseConfiguration` which would contain all those fields like `Username`, `Host` etc. It looks kind of weird to have all those `DB...`.

File: backend/config/config.go [hunk 3]
```diff
@@ -10,37 +10,22 @@ import (
...
+	DBPassword string
+	DBHost     string
+	DBPort     string
+
+	redisPassword string
```
Comment (VirrageS) [inline backend/config/config.go]: I think that there should be new struct `CacheConfiguration` which would contain all those fields like `Password`, `Host` etc.

Field `cacheExpirationTime` should be also included into this new struct.

File: backend/config.yaml [hunk 4]
```diff
@@ -3,3 +3,15 @@ auth_token_validity_period: 15
...
 
 cache_expiration_time: 1m
+
+database:
+  username: "postgres"
+  password: "postgres"
+  host: "localhost"
+  port: "5432"
+
+redis:
```
Comment (VirrageS) [inline backend/config.yaml]: `cache` instead of `redis`

File: backend/database/database.go [hunk 5]
```diff
@@ -26,11 +26,14 @@ func NewDatabase(databaseConnection *sql.DB, cache cache.CacheProvider) Database
...
-	// TODO: read user data, host and port from config file
-
-	db, err := sql.Open("postgres", "user=postgres password=postgres host=localhost sslmode=disable port="+port)
+func NewConnection(config config.DBAccessConfigProvider) *sql.DB {
+	username := config.GetDBUsername()
+	password := config.GetDBPassword()
+	host := config.GetDBHost()
+	port := config.GetDBPort()
+
+	dbAccesString := fmt.Sprintf("user=%s password=%s host=%s sslmode=disable port=%s", username, password, host, port)
```
Comment (VirrageS) [inline backend/database/database.go]: `dbAccesString` -> `dbAccessString` or just `accessString`.

File: backend/config/interfaces.go (hunk 1 above)
Comment (Prytu) [inline backend/config/interfaces.go]: yes, but are we sure that memcache etc uses the same parameters as redis?

File: backend/config.yaml (hunk 4 above)
Comment (Prytu) [inline backend/config.yaml]: I'd stay with redis because at the moment we use redis which has got its own parameters

**Pull request merged**